cardinal/chain/adapter.go

```diff
@@ -17,8 +18,18 @@ import (
 
 // Adapter is a type that helps facilitate communication with the EVM base shard.
 type Adapter interface {
-	Submit(ctx context.Context, namespace string, tick uint64, txs []byte) error
-	QueryBatches(ctx context.Context, req *shardtypes.QueryBatchesRequest) (*shardtypes.QueryBatchesResponse, error)
+	Writer
+	Reader
+}
+
+type Writer interface {
+	Submit(ctx context.Context, p *sign.SignedPayload, txID, tick uint64) error
```

Comment:
can we comment on what this is needed for/does

Reply:
done